packages/prisma-generator-proto/src/generate/hooks.ts

```diff
@@ -28,122 +21,150 @@ function getFieldTsType(field: DMMF.Field): string {
   return fieldMetadata.type;
 }
 
+/**
+ * Calculate the TS type of the model's ID.
+ */
+function getIdTsType(model: DMMF.Model): string {
+  for (const field of model.fields) {
+    if (field.isId) return getFieldTsType(field);
+  }
+  return "never";
+}
+
+/**
+ * @returns `true` if creating a row will fail without specifying the given field.
+ */
+function needsInitialValue(field: DMMF.Field): boolean {
```

Comment:
`hooks.ts` file is growing quickly. In future iteration consider:
- adding unit tests.
- adding docs.
- splitting the hook generation logic into smaller modules.